Libraries/Experimental/SwipeableRow/SwipeableRow.js

```diff
@@ -59,79 +57,57 @@ const RIGHT_SWIPE_THRESHOLD = 30 * SLOW_SPEED_SWIPE_FACTOR;
 type Props = $ReadOnly<{|
   children?: ?React.Node,
   isOpen?: ?boolean,
-  maxSwipeDistance?: ?number,
-  onClose?: ?Function,
-  onOpen?: ?Function,
-  onSwipeEnd?: ?Function,
-  onSwipeStart?: ?Function,
+  maxSwipeDistance: number,
+  onClose: Function,
+  onOpen: Function,
+  onSwipeEnd: Function,
+  onSwipeStart: Function,
   preventSwipeRight?: ?boolean,
   shouldBounceOnMount?: ?boolean,
   slideoutView?: ?React.Node,
-  swipeThreshold?: ?number,
+  swipeThreshold: number,
 |}>;
 
+type State = {
+  currentLeft: Animated.Value,
+  isSwipeableViewRendered: boolean,
+  rowHeight: ?number,
+};
+
 /**
  * Creates a swipable row that allows taps on the main item and a custom View
  * on the item hidden behind the row. Typically this should be used in
  * conjunction with SwipeableListView for additional functionality, but can be
  * used in a normal ListView. See the renderRow for SwipeableListView to see how
  * to use this component separately.
  */
-const SwipeableRow = createReactClass({
-  displayName: 'SwipeableRow',
-  _panResponder: {},
-  _previousLeft: CLOSED_LEFT_POSITION,
-  _timeoutID: (null: ?TimeoutID),
-
-  propTypes: {
-    children: PropTypes.any,
-    isOpen: PropTypes.bool,
-    preventSwipeRight: PropTypes.bool,
-    maxSwipeDistance: PropTypes.number.isRequired,
-    onOpen: PropTypes.func.isRequired,
-    onClose: PropTypes.func.isRequired,
-    onSwipeEnd: PropTypes.func.isRequired,
-    onSwipeStart: PropTypes.func.isRequired,
-    // Should bounce the row on mount
-    shouldBounceOnMount: PropTypes.bool,
-    /**
-     * A ReactElement that is unveiled when the user swipes
-     */
-    slideoutView: PropTypes.node.isRequired,
+class SwipeableRow extends React.Component<Props, State> {
+  _panResponder = {};
+  _previousLeft = CLOSED_LEFT_POSITION;
+  _timeoutID: ?TimeoutID = null;
+
+  static defaultProps = {
+    isOpen: false,
+    preventSwipeRight: false,
+    maxSwipeDistance: 0,
+    onOpen: emptyFunction,
+    onClose: emptyFunction,
+    onSwipeEnd: emptyFunction,
+    onSwipeStart: emptyFunction,
+    swipeThreshold: 30,
+  };
```

Comment:
The problem with `defaultProps` is that it only defaults `undefined` props (and ignores when props are `null`). To make a component's API flexible, we should also default `null` props. This is why all of the optional props in this component are also nullable. However, the typings in this case are actually incorrect because we don't default nullable props. For example, `maxSwipeDistance` is nullable, so `defaultProps` doesn't default it to 0, but everywhere in the code, we assume that it's been defaulted to a number. 

Instead of relying on `defaultProps`, we should probably use the `??` operator:

```JavaScript
_animateToOpenPosition(): void {
  const maxSwipeDistance = this.props.maxSwipeDistance ?? 0;
  const directionAwareMaxSwipeDistance = IS_RTL ? -maxSwipeDistance : maxSwipeDistance;
  this._animateTo(-directionAwareMaxSwipeDistance);
},
```

What do you think about getting rid of `defaultProps` and using the `??` operator, which has this definition:
```JavaScript
  x ?? y === x != null ? x : y
```